tests/test_types.py

```diff
@@ -3329,8 +3429,30 @@ class Model(BaseModel):
         (Set[int], [{1, 2}, {3, 4}, {5, 6}], deque([{1, 2}, {3, 4}, {5, 6}])),
         (Tuple[int, str], ((1, 'a'), (2, 'b'), (3, 'c')), deque(((1, 'a'), (2, 'b'), (3, 'c')))),
         (str, [w for w in 'one two three'.split()], deque(['one', 'two', 'three'])),
-        # (float, {1.0, 2.0, 3.0}, deque([1.0, 2.0, 3.0])),
-        # (int, frozenset([1, 2, 3]), deque([1, 2, 3])),
+        pytest.param(
+            int,
+            {1: 10, 2: 20, 3: 30}.keys(),
+            deque([1, 2, 3]),
+            marks=pytest.mark.skipif(
+                condition=python_implementation() == 'PyPy', reason='PyPy does not implement a dict_keys type'
+            ),
+        ),
+        pytest.param(
+            int,
+            {1: 10, 2: 20, 3: 30}.values(),
+            deque([10, 20, 30]),
+            marks=pytest.mark.skipif(
+                condition=python_implementation() == 'PyPy', reason='PyPy does not implement a dict_values type'
+            ),
+        ),
+        pytest.param(
+            Tuple[int, int],
+            {1: 10, 2: 20, 3: 30}.items(),
+            deque([(1, 10), (2, 20), (3, 30)]),
+            marks=pytest.mark.skipif(
+                condition=python_implementation() == 'PyPy', reason='PyPy does not implement a dict_items type'
+            ),
+        ),
```

Comment:
I had to skip these tests on PyPy. I bet that if we replaced https://github.com/pydantic/pydantic-core/blob/7a4666490d825dd766dc98f506be5ea526eb5cfb/src/input/input_python.rs#L11 with a GILOnceCell that gets `type({}.keys())` and such it would work on PyPy.